.github/workflows/workflow.yml

```diff
@@ -0,0 +1,50 @@
+name: Main workflow
+
+on:
+  - pull_request
+#  - push
+
+jobs:
+  build_and_test:
+    name: Build and test
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
+        ocaml-compiler:
+          - 4.05.x
+          - 4.08.x
+          - 4.13.x
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+          opam-local-packages: hamt.opam
+
+      - run: opam install --deps-only --with-doc --with-test hamt
+
+      - run: opam exec -- dune build -p hamt
```

Comment:
I'd remove the `-p` here. As otherwise it wouldn't fail on warnings.

Reply:
I replaced -p with --only-packages (that's still needed because we don't want to build hamt-bench here since it will fail on 4.05, and just removing the -p makes dune fail because it doesn't know what to do).